dockstore-client/src/main/java/io/dockstore/client/cli/YamlClient.java

```diff
@@ -0,0 +1,108 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.dockstore.client.cli.ArgumentUtility.out;
+import static io.dockstore.client.cli.JCommanderUtility.printJCommanderHelp;
+import static io.dockstore.client.cli.JCommanderUtility.printJCommanderHelpYaml;
+
+/**
+ *
+ */
+public final class YamlClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YamlClient.class);
+
+
+    private YamlClient() {
+        // disable constructor for utility class
+    }
+
+
+    /**
+     * @param args
+     */
+    public static boolean handleCommand(List<String> args) {
+        String[] argv = args.toArray(new String[args.size()]);
+        JCommander jc = new JCommander();
+
+        CommandYaml commandYaml = new CommandYaml();
+        JCommander jcPlugin = JCommanderUtility.addCommand(jc, "yaml", commandYaml);
+        CommandYamlValidate commandYamlValidate = new CommandYamlValidate();
+        commandYamlValidate = new CommandYamlValidate();
+        JCommanderUtility.addCommand(jcPlugin, "validate", commandYamlValidate);
+
+        // Not parsing with jc because we know the first command was plugin.  jc's purpose is to display help
+        jcPlugin.parse(argv);
+        try {
+            if (args.isEmpty() || commandYaml.help) {
+                printJCommanderHelpYaml(jc, "dockstore", "yaml");
+            } else {
+                switch (jcPlugin.getParsedCommand()) {
+                case "validate":
+                    if (commandYamlValidate.help) {
+                        printJCommanderHelpYaml(jc, "dockstore", "yaml");
+                    } else if (commandYamlValidate.path == null) {
+                        printJCommanderHelpYaml(jc, "dockstore", "yaml");
+                        out("ERROR: Missing --path <path>");
+                    } else {
+                        try {
+                            YamlVerify.dockstoreValidate(CommandYamlValidate.path);
+                        } catch (ValidateYamlException Ex) {
+                            out(Ex.getMessage());
+                        }
+                        return true;
+                    }
+                    break;
+                default:
+                    // fall through
+                }
+            }
+        } catch (ParameterException e) {
+            printJCommanderHelp(jc, "dockstore", "yaml");
+        }
+        return true;
+
+    }
+
+
+    @Parameters(separators = "=", commandDescription = "Tools used for yaml files")
+    private static class CommandYaml {
+        @Parameter(names = "--help", description = "Prints help for yaml command", help = true)
+        private boolean help = false;
+
+    }
+
+    @Parameters(separators = "=", commandDescription = "List currently activated file provision plugins")
+    private static class CommandYamlValidate {
+        @Parameter(names = "--path", description = "Complete entry path on computer (ex. /home/usr/test)", required = false)
```

Comment:
a follow-up ticket can be created for relative paths (a little more convenient, but not must-have)